### ISSUE TITLE
Lots and lots of duplicate scrobbles

### ISSUE THREAD

**Issue by matrixagent** [trakt/Plex-Trakt-Scrobbler#315]
Hi @fuzeman,

I've been using the new version since about one or two weeks after it came out of beta. Since then, I've been closely monitoring my trakt activity and I noticed a huge increase in duplicates on my trakt history. It's not for every item I watch, but I'd say roughly two in three times I'll have to manually fix a duplicate scrobbled entry. The entries are usually within 5 minutes of each other, so it seems like it might scrobble once when it reaches the threshold (is it still 90 or 75 percent of the running time?) and then again upon actual completion.

<img width="989" alt="nwdkr2016easp0055v4ni15lbym919oo42l39" src="https://cloud.githubusercontent.com/assets/309726/15275683/59d5d8a0-1ad2-11e6-9fef-e018563f4682.png">

As it does not happen all the time I figure it's not a configuration issue. Any idea what might cause this and/or what I could check for in the logs to get to the bottom of this?

**Comment by spyzetetic** [comment]
I have the same issue, lots of duplicate scrobbles but it only occurs when I use the any DLNA client via the Plex DLNA server, not when I use the Plex app on my PS4. The duplicates aren't evenly spaced in time though, sometimes as many as 2 or 3 every minute, sometimes every 2-3 minutes. I have a QNAP as my NAS.

**Comment by matrixagent** [comment]
To clarify my specific issue with regard to your additions, @spyzetetic: I'm using RasPlex (e.g. openPHT) and the official iOS apps for playback. No DLNA. Also, I'm really only getting duplicates, not more than two multiple scrobbles per one episode.

In general, I think the plugin would benefit from better checking for such duplicates. I can be certain to get a duplicate when I pause an hour-long episode five minutes before the end and then continue playback a few minutes later. However, I think it could safely be assumed that there will never be a case in which I'd want two scrobbles for an item that is 30+ minutes long within a period that is shorter than, e.g. 50% of that time.

**Comment by rectifyer** [comment]
I'm going to look into this server side too since it makes sense for the Trakt API to ignore duplicate scrobbles if its the same exact item being scrobbled. A combination of plugin and server side seems like it they would both help clean this up.

**Comment by fuzeman** [comment]
I'm aware of a bug where items can sometimes be scrobbled twice if a sync runs while you are watching something, will work on a fix for that issue this week.

------------------------------

@spyzetetic DLNA client scrobbling has been broken for possibly over a year now, this was originally caused by an update to Plex Media Server that resulted in DLNA clients sending invalid "timeline" (progress) messages.

This might of been fixed in a recent PMS update, will check this is still the case.

------------------------------------

@matrixagent 
> In general, I think the plugin would benefit from better checking for such duplicates. I can be certain to get a duplicate when I pause an hour-long episode five minutes before the end and then continue playback a few minutes later. However, I think it could safely be assumed that there will never be a case in which I'd want two scrobbles for an item that is 30+ minutes long within a period that is shorter than, e.g. 50% of that time.

We do [check for duplicates](https://github.com/trakt/Plex-Trakt-Scrobbler/blob/master/Trakttv.bundle/Contents/Libraries/Shared/plugin/managers/action.py#L157) during scrobbling, but for some reason that method allows duplicates when media reaches 80% progress.. will make some changes to that method later this week.

------------------------------------------

@rectifyer 
Sounds good, that will definitely help.

Will you check for duplicates added with the ```/sync/history``` method? or would this only apply to the ```/scrobble/stop``` method?

**Comment by rectifyer** [comment]
@fuzeman this will only apply to `scrobble/stop` for now. We need to complete some backend performance updates before we can apply a similar update to `sync/history`. In that case, the plan is to ignore additional plays with the same timestamp.

**Comment by matrixagent** [comment]
I'm always amazed by how professional you guys are handling these projects. 👍 Thanks for the quick and positive reply, especially appreciate your direct reference to relevant lines of code, @fuzeman!

I think it is very likely that my problem is caused by what you mentioned first, as I've also noticed my syncs are taking very, very long. But I'll make another ticket for that.

**Comment by fuzeman** [comment]
I've made a few improvements in [v1.0.2.1](https://github.com/trakt/Plex-Trakt-Scrobbler/releases/tag/v1.0.2.1) that should reduce the chances of scrobble duplication.

Duplicate scrobbles (for the same movie or episode) that happen in the same 6-hour period will now be ignored, this can be adjusted (or disabled) with the `Scrobble - Ignore duplicates for` option.
